### PR TITLE
Reminders

### PR DESCRIPTION
**Pull request by tjerman**
Add v1 for reminder service.

### CODE REVIEW

File: api/system/spec.json [hunk 1]
```diff
@@ -1593,5 +1593,206 @@
...
+      }
+    ],
+    "apis": [
+      {
+        "name": "list",
+        "method": "GET",
+        "title": "List/read reminders",
+        "path": "/",
+        "parameters": {
+          "get": [
```
[the remark this answers is not in the document]
Comment (tjerman) [reply api/system/spec.json]: I'm guessing that by 'perPage=0' you mean no limit?

[the remark this answers is not in the document]
Comment (darh) [reply api/system/spec.json]: yes, in context of fetching (all) future reminders.

File: system/repository/reminder.go [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+func (r reminder) Find(filter types.ReminderFilter) (set types.ReminderSet, err error) {
+	q := r.query()
+
+	if filter.AssignedTo != 0 {
+		q = q.Where("r.assigned_to = ?", filter.AssignedTo)
+	}
+
+	if filter.Resource != "" {
+		q = q.Where("r.resource LIKE ?%", filter.Resource)
```
[the remark this answers is not in the document]
Comment (tjerman) [reply system/repository/reminder.go]: > b) is resource-lookup-by-prefix really what we want? I can not think of a scenario where we need it

It seemed like a nice to have feature.

File: system/types/reminder.go [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+		CreatedAt time.Time  `json:"createdAt,omitempty" db:"created_at"`
+		UpdatedAt *time.Time `json:"updatedAt,omitempty" db:"updated_at"`
+		DeletedAt *time.Time `json:"deletedAt,omitempty" db:"deleted_at"`
+	}
+
+	ReminderFilter struct {
+		Resource   string `json:"resource"`
+		AssignedTo uint64 `json:"assignedTo,uint64"`
+		Page       uint   `json:"page,uint"`
```
[the remark this answers is not in the document]
Comment (tjerman) [reply system/types/reminder.go]: I realized that a bit late; my bad. Working on it

File: system/repository/reminder.go [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+	}
+
+	// @todo allow sorting at some point
+	q = q.OrderBy("r.remind_at")
+
+	return set, r.fetchPaged(&set, q, filter.Page, filter.PerPage)
+}
+
+func (r reminder) FindByID(ID uint64) (rm *types.Reminder, err error) {
```
[the remark this answers is not in the document]
Comment (darh) [reply system/repository/reminder.go]: true &sorry -- compose module:
https://github.com/cortezaproject/corteza-server/blob/imp-exp/compose/repository/module.go#L87

File: system/service/reminder.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+		return nil, ErrNoUpdatePermissions
+	}
+
+	return rm, svc.db.Transaction(func() (err error) {
+		if t, err = svc.reminder.FindByID(rm.ID); err != nil {
+			return
+		}
+
+		// Assign changed values
+		if rm.AssignedTo != t.AssignedTo {
```
[the remark this answers is not in the document]
Comment (tjerman) [reply system/service/reminder.go]: Currently no, since create stores whatever reminder's request handler prepares

**Pull request closed without merging**